### PR TITLE
Reduce dependencies

### PR DESCRIPTION
**Pull request by isidentical**
more to come

### CODE REVIEW

File: projects/fal/pyproject.toml [hunk 1]
```diff
@@ -41,7 +40,6 @@ types-python-dateutil = "^2.8.0"
 
 # For 3.9 and earlier, importlib-metadata's newer versions are included in the standard library.
 importlib-metadata = { version = ">=4.4", python = "<3.10" }
-boto3 = "^1.33.8"
```
Comment (isidentical) [inline projects/fal/pyproject.toml:44]: this is only used in r2 implementation, and we don't really like to bring it down with us (just like google-cloud-storage). cc: @badayvedat

File: projects/fal/pyproject.toml [hunk 2]
```diff
@@ -14,7 +14,6 @@ python = "^3.8"
 isolate = {version = ">=0.12.3, <1.0", extras = ["build"]}
 isolate-proto = {path = "../isolate_proto", develop = true}
 auth0-python = "^4.1.0"
```
Comment (isidentical) [inline projects/fal/pyproject.toml:16]: We only seem to use this to re-validate the token we receive, is that simply a debug assertion @chamini2? Can we drop `validate_id_token` and with it, `auth0` dependency?

**Review by isidentical — COMMENTED** [review]
we agreeed to drop client-side tracing, so i can probably drop `opentelemetry-api`/`opentelemetry-sdk` with it.

File: projects/fal/pyproject.toml (hunk 1 above)
Comment (badayvedat) [reply projects/fal/pyproject.toml:44]: makes sense! let's remove it

**Review by mederka — APPROVED** [review]

File: projects/fal/pyproject.toml (hunk 2 above)
Comment (chamini2) [reply projects/fal/pyproject.toml:16]: looking into it

**Pull request merged**